appveyor.yml

```diff
@@ -0,0 +1,51 @@
+environment:
+  PATH: C:\msys64\usr\bin;C:\msys64\mingw64\bin;C:\Windows\System32;C:\Windows;%PATH%
+  VSTEST: OFF
+  
+  matrix:
+    # x86 builds
+    - GENERATOR: Visual Studio 14 2015
+      CONFIG: Release
+      SHARED_LIBS: ON
+    - GENERATOR: Visual Studio 14 2015
+      CONFIG: Release
+      SHARED_LIBS: OFF
+    - GENERATOR: Visual Studio 14 2015
+      CONFIG: Debug
+      SHARED_LIBS: ON
+    - GENERATOR: Visual Studio 14 2015
+      CONFIG: Debug
+      SHARED_LIBS: ON
+
+    # x64 builds
+    - GENERATOR: Visual Studio 14 2015 Win64
+      CONFIG: Release
+      SHARED_LIBS: ON
+    - GENERATOR: Visual Studio 14 2015 Win64
+      CONFIG: Release
+      SHARED_LIBS: OFF
+    - GENERATOR: Visual Studio 14 2015 Win64
+      CONFIG: Debug
+      SHARED_LIBS: ON
+    - GENERATOR: Visual Studio 14 2015 Win64
+      CONFIG: Debug
+      SHARED_LIBS: ON
+
+init:
+  # update mysy2
+  - C:\msys64\usr\bin\bash -lc "pacman --needed --noconfirm -Sy pacman-mirrors"
+  - C:\msys64\usr\bin\bash -lc "pacman --noconfirm -Sy"
+  - C:\msys64\usr\bin\bash -lc "pacman --noconfirm -S autoconf perl"
+  - C:\msys64\usr\bin\bash -lc "pacman --noconfirm -S git"
```

Comment:
I couldn't get LibreSSL to build with the stock mingw that comes with its git. I saw that some projects actually updated msys64 so I went ahead and did that so the `autogen.sh` script could be run.